lib/pbio/drv/sound/sound_ev3.c

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025 The Pybricks Authors
+
+#include <pbdrv/config.h>
+
+#if PBDRV_CONFIG_SOUND_EV3
+
+#include <stdint.h>
+
+#include <pbdrv/gpio.h>
+
+#include <tiam1808/ehrpwm.h>
+#include <tiam1808/hw/soc_AM1808.h>
+#include <tiam1808/hw/hw_syscfg0_AM1808.h>
+#include <tiam1808/hw/hw_types.h>
+#include <tiam1808/psc.h>
+
+#include "../drv/gpio/gpio_ev3.h"
+
+// The division factor of 40 gives decent coverage of the frequency range
+// 64 Hz - 24 kHz that the other drivers support.
+// (This is limited by the 16-bit period counter.)
+#define PWM_TIMEBASE_FREQ       (SOC_EHRPWM_0_MODULE_FREQ / 40)
```

Comment:
Not necessarily for this pull request, but we need to have a look at the clock tree.

In the official LEGO firmware and in ev3dev, we have the ASYNC3 clock mux selecting SYSCLK2 from PLL1 rather than PLL2. It is 132 MHz instead of 150MHz.